src/RequestParser.php

```diff
@@ -28,12 +28,40 @@ public function __construct(callable $readParameter, $config)
             $config = new Config();
             $config->setExceptionMessageFactory(new LegacyExceptionMessageFactory());
             $config->setExceptionFactory(new LegacyExceptionFactory($exceptionFactory));
+
+            trigger_error($this->generateCallbackBasedExceptionHandlingDeprecatedMessage($exceptionFactory), E_USER_DEPRECATED);
         }
 
         $this->readParameter = $readParameter;
         $this->config = $config;
     }
 
+    private function generateCallbackBasedExceptionHandlingDeprecatedMessage(callable $callback)
+    {
+        $testException = $callback('$parameterName');
+
+        $exceptionClass = get_class($testException);
+        $exceptionMessage = $testException->getMessage();
+
+        $callbackCode = "function (\$parameterName) { return new $exceptionClass(\"...\"); }";
+
+        $message = "RequestParser: Callback based Exception configuration is deprecated. ";
+        $message .= "To get get better error messages for invalid parameters (e.g. a string is provided to a `->int()->required()`) ";
+        $message .= "you just need to migrate your code (more infos: http://bit.ly/2aGyX99).\n\n";
```

Comment:
Short url to keep this text under 1024 chars (that's the hard limit of `trigger_error`)